### PR TITLE
Fix line numbers in checkstyle output

### LINKED ISSUE #1386: checkstyle line numbers are offset
in src/rustfmt_diff.rs line 41   line number is offset by context_queue to start the line at the start of the patch

      mismatch = Mismatch::new(line_number - context_queue.len() as u32);

but will make the output of checkstyle incorrect.
this affects PR1384

would adding be another member to Mismatch with the real line number be ok?

### PR DESCRIPTION
**Pull request by calebcartwright**
Resolves #1386

### CODE REVIEW

File: src/emitter/checkstyle.rs [hunk 1]
```diff
@@ -52,8 +52,11 @@ where
             if let DiffLine::Expected(message) = line {
                 write!(
                     writer,
-                    r#"<error line="{}" severity="warning" message="Should be `{}`" />"#,
+                    r#"<error {}="{}" {}="{}" severity="warning" message="Should be `{}`" />"#,
```
Comment (calebcartwright) [inline src/emitter/checkstyle.rs]: Embedding all of the attribute names directly in the template literal was resulting in it exceeding the limit, causing the `line formatted, but exceeded maximum width (maximum: 100 (see `max_width` option), found: 108)` error, hence this change to the template

**Comment by calebcartwright** [comment]
Actually there's more going wrong here than I realized 🤔 

The current checkstyle output generated from formatting [`tests/writemode/source/fn-single-line.rs`](https://github.com/rust-lang/rustfmt/blob/master/tests/writemode/source/fn-single-line.rs) is way off on the calcs for both the `line_number` and `line_number_orig` values in the `Mismatch` :

```xml
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle version="4.3">
  <file name="tests/writemode/source/fn-single-line.rs">
    <error line="2" original_line="2" severity="warning" message="Should be `fn foo_expr() { 1 }`" />
    <error line="2" original_line="2" severity="warning" message="Should be `fn foo_stmt() { foo(); }`" />
    <error line="2" original_line="2" severity="warning" message="Should be `fn foo_decl_local() { let z = 5; }`" />
    <error line="2" original_line="2" severity="warning" message="Should be `fn foo_decl_item(x: &amp;mut i32) { x = 3; }`" />
    <error line="2" original_line="2" severity="warning" message="Should be `fn empty() {}`" />
    <error line="2" original_line="2" severity="warning" message="Should be `fn foo_return() -&gt; String { &quot;yay&quot; }`" />
    <error line="2" original_line="2" severity="warning" message="Should be `fn foo_where() -&gt; T`" />
    <error line="2" original_line="2" severity="warning" message="Should be `where`" />
    <error line="2" original_line="2" severity="warning" message="Should be `    T: Sync,`" />
    <error line="2" original_line="2" severity="warning" message="Should be `{`" />
    <error line="52" original_line="61" severity="warning" message="Should be `fn lots_of_space() { 1 }`" />
    <error line="59" original_line="70" severity="warning" message="Should be `    fn dummy(&amp;self) {}`" />
    <error line="59" original_line="70" severity="warning" message="Should be `trait CoolerTypes {`" />
    <error line="59" original_line="70" severity="warning" message="Should be `    fn dummy(&amp;self) {}`" />
    <error line="59" original_line="70" severity="warning" message="Should be `fn Foo&lt;T&gt;()`" />
    <error line="59" original_line="70" severity="warning" message="Should be `where`" />
    <error line="59" original_line="70" severity="warning" message="Should be `    T: Bar,`" />
    <error line="59" original_line="70" severity="warning" message="Should be `{`" />
  </file>
</checkstyle>
```

**Comment by scampi** [comment]
both original and formatted line numbers are off...

**Comment by calebcartwright** [comment]
I decided to keep things simple here, and just address the offset line numbers. Below is an example of the checkstyle output that is now emitted (pretty printed for ease of view). See above for what this same output looked like before. 

My recommendation is that anyone wanting a more complex output (line numbers from the before/after versions of the file, start/end line numbers, etc.) should instead use the new json emit mode added in https://github.com/rust-lang/rustfmt/pull/3735 as the json format should be easier to work with.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle version="4.3">
  <file name="tests/writemode/source/fn-single-line.rs">
    <error line="5" severity="warning" message="Should be `fn foo_expr() { 1 }`" />
    <error line="7" severity="warning" message="Should be `fn foo_stmt() { foo(); }`" />
    <error line="9" severity="warning" message="Should be `fn foo_decl_local() { let z = 5; }`" />
    <error line="11" severity="warning" message="Should be `fn foo_decl_item(x: &amp;mut i32) { x = 3; }`" />
    <error line="13" severity="warning" message="Should be `fn empty() {}`" />
    <error line="15" severity="warning" message="Should be `fn foo_return() -&gt; String { &quot;yay&quot; }`" />
    <error line="17" severity="warning" message="Should be `fn foo_where() -&gt; T`" />
    <error line="18" severity="warning" message="Should be `where`" />
    <error line="19" severity="warning" message="Should be `    T: Sync,`" />
    <error line="20" severity="warning" message="Should be `{`" />
    <error line="55" severity="warning" message="Should be `fn lots_of_space() { 1 }`" />
    <error line="60" severity="warning" message="Should be `    fn dummy(&amp;self) {}`" />
    <error line="63" severity="warning" message="Should be `trait CoolerTypes {`" />
    <error line="64" severity="warning" message="Should be `    fn dummy(&amp;self) {}`" />
    <error line="67" severity="warning" message="Should be `fn Foo&lt;T&gt;()`" />
    <error line="68" severity="warning" message="Should be `where`" />
    <error line="69" severity="warning" message="Should be `    T: Bar,`" />
    <error line="70" severity="warning" message="Should be `{`" />
  </file>
</checkstyle>
```

**Comment by topecongiro** [comment]
Sorry for the late review on this one. Thank you for your works!

**Comment by calebcartwright** [comment]
> Sorry for the late review on this one. Thank you for your works!

No worries, and my pleasure! I'm still pretty new to Rust but I find working with rustfmt is both fun and helping me learn. I'm keeping an eye out for any issues I think I can help with, but please also let me know if there's anything in particular I can do to help!

**Pull request merged**